src/version/mod.rs

```diff
@@ -637,3 +637,87 @@ impl Iterator for LevelFileNumIterator {
         Ok(())
     }
 }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::format::{InternalKey, InternalKeyComparator, ValueType};
+    use crate::util::comparator::BytewiseComparator;
+    use crate::util::slice::Slice;
+
+    struct FindFileTests {
+        pub files: Vec<Arc<FileMetaData>>,
+        cmp: Arc<InternalKeyComparator>,
+    }
+
+    impl FindFileTests {
+        fn new() -> Self {
+            let files: Vec<Arc<FileMetaData>> = Vec::new();
+            let cmp = Arc::new(InternalKeyComparator::new(Arc::new(
+                BytewiseComparator::new(),
+            )));
+
+            Self { files, cmp }
+        }
+
+        fn generate(&mut self, smallest: &Slice, largest: &Slice) {
+            let mut file = FileMetaData::default();
+            file.number = self.files.len() as u64 + 1;
+            file.smallest = Rc::new(InternalKey::new(smallest, 100, ValueType::Value));
+            file.largest = Rc::new(InternalKey::new(largest, 100, ValueType::Value));
+            self.files.push(Arc::new(file));
+        }
+
+        fn find(&self, key: &Slice) -> usize {
+            let target = Slice::from(InternalKey::new(key, 100, ValueType::Value).data());
```

Comment:
The `InternalKey::new(key, 100, ValueType::Value).data()` doesn't  be alive as long as the outer `Slice` which will make `target` become a UB.

Reply:
@AliceLanniste Finally I found this 😂